src/json.hpp

```diff
@@ -2702,32 +2644,39 @@ class lexer
     // input management
     /////////////////////
 
-    /// reset yytext
+    /// reset yytext; current character is beginning of token
     void reset() noexcept
     {
-        yylen = 0;
-        start_pos = chars_read - 1;
+        yytext.clear();
+        token_string.clear();
+        token_string.push_back(static_cast<char>(current));
     }
 
     /// get a character from the input
     int get()
     {
         ++chars_read;
-        return next_unget ? (next_unget = false, current)
-               : (current = ia->get_character());
+        int c = current = ia->get_character();
+        token_string.push_back(static_cast<char>(c));
```

Comment:
Could you only use `current` here? By the way I think we should have a convention about data members, either explicitly use `this->`, or an underscore prefix etc...

But that's not in the PR's scope.

Reply:
I'm not a fan of neither namings...

Reply:
I do not really care about which naming to use, but I get confused each time, is it a local variable? A global one? A symbol in the namespace?

Reply:
(Sorry, I'm ok with only using `current` - I meant I don't like adding an underscore prefix or `this->`)

Reply:
I understand, it's just hard to know what the symbol is without performing a search in the current file.